python/semantic_kernel/connectors/memory/azure_cosmos_db/azure_cosmos_db_no_sql_collection.py

```diff
@@ -0,0 +1,212 @@
+    MemoryConnectorException,
+    MemoryConnectorResourceNotFound,
+    VectorStoreModelDeserializationException,
+)
+from semantic_kernel.kernel_types import OneOrMany
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+TModel = TypeVar("TModel")
+TKey = TypeVar("TKey", str, AzureCosmosDBNoSQLCompositeKey)
+
+
+@experimental_class
+class AzureCosmosDBNoSQLCollection(AzureCosmosDBNoSQLBase, VectorStoreRecordCollection[TKey, TModel]):
+    """An Azure Cosmos DB NoSQL collection stores documents in a Azure Cosmos DB NoSQL account."""
+
+    partition_key: PartitionKey
+
+    def __init__(
+        self,
+        data_model_type: type[TModel],
+        database_name: str,
+        collection_name: str,
+        data_model_definition: VectorStoreRecordDefinition | None = None,
+        url: str | None = None,
+        key: str | None = None,
+        cosmos_client: CosmosClient | None = None,
+        partition_key: PartitionKey | None = None,
+        create_database: bool = False,
+    ):
+        """Initializes a new instance of the AzureCosmosDBNoSQLCollection class.
+
+        Args:
+            data_model_type (type[TModel]): The type of the data model.
+            database_name (str): The name of the database. Used to create a database proxy if not provided.
+            collection_name (str): The name of the collection.
+            data_model_definition (VectorStoreRecordDefinition): The definition of the data model. Defaults to None.
+            url (str): The URL of the Azure Cosmos DB NoSQL account. Defaults to None.
+            key (str): The key of the Azure Cosmos DB NoSQL account. Defaults to None.
+            cosmos_client (CosmosClient): The custom Azure Cosmos DB NoSQL client whose lifetime is managed by the user.
+            partition_key (PartitionKey): The partition key. Defaults to None. If not provided, the partition
+                                          key will be based on the key field of the data model definition.
+            create_database (bool): Indicates whether to create the database if it does not exist.
+                                    Defaults to False.
+        """
+        super().__init__(
+            database_name=database_name,
+            url=url,
+            key=key,
+            cosmos_client=cosmos_client,
+            create_database=create_database,
+            data_model_type=data_model_type,
+            data_model_definition=data_model_definition,
+            collection_name=collection_name,
+            partition_key=partition_key or PartitionKey(path=f"/{COSMOS_ITEM_ID_PROPERTY_NAME}"),
+        )
+
+    @override
+    async def _inner_upsert(
+        self,
+        records: Sequence[Any],
+        **kwargs: Any,
+    ) -> Sequence[TKey]:
+        try:
+            container_proxy = await self._get_container_proxy(self.collection_name)
+            results = await asyncio.gather(*[container_proxy.upsert_item(body=record) for record in records])
```

Comment:
there is a batch command (`execute_item_batch`) which could be usefull when we can figure out which items go to the same partition key value